lib/src/views/home/widgets/recipe_list.dart

```diff
@@ -0,0 +1,298 @@
+import 'package:flutter/material.dart';
+import 'package:shimmer/shimmer.dart';
+import 'package:web_provise_test/src/apis/recipe/recipe_api_impl.dart';
+import 'package:web_provise_test/src/models/recipe.dart';
+import 'package:web_provise_test/src/utils/extensions.dart';
+import 'package:web_provise_test/src/views/home/widgets/recipe_card.dart';
+import 'package:web_provise_test/src/views/home/widgets/recipe_detail.dart';
+
+class RecipeList extends StatefulWidget {
+  const RecipeList({super.key});
+
+  @override
+  State<RecipeList> createState() => _RecipeListState();
+}
+
+class _RecipeListState extends State<RecipeList> {
+  final RecipeApiImpl _recipeApi = RecipeApiImpl();
+  List<Recipe> recipes = [];
+  bool isLoading = true;
+
+  @override
+  void initState() {
+    super.initState();
+    fetchData();
+  }
+
+  Future<void> fetchData() async {
```

Comment:
All properties that are not gonna exposed for outside use should define as private.
`_ recipes, _isLoading, _fetchData`